old_projects/hilbert/section1.py

```diff
@@ -659,7 +659,7 @@ def construct(self):
         mathy, bubble = get_mathy_and_bubble()
         squiggle_mouth = mathy.mouth.copy()
         squiggle_mouth.apply_function(
-            lambda (x, y, z) : (x, y+0.02*np.sin(50*x), z)
+            lambda x_y_z : (x_y_z[0], x_y_z[1]+0.02*np.sin(50*x_y_z[0]), x_y_z[2])
```

Comment:
Is this more pythonic?

Reply:
It is not about being Pythonic.  It is about avoiding syntax errors.  Putting the parameters to a lambda in parens is a syntax error in Python 3.

Reply:
Multiple arguments to a lambda isn't a syntax error if you just remove the parentheses:
    >>> func = lambda x,y,z: x+x
    >>> func(1,2,3)
    2

Reply:
But then there are multiple places in code that expect that this lambdas need tuples as parameters. It could be change later if only thing is better python 3 compatibility without changing workflow. @3b1b?

Reply:
There are two ways that the caller can call the lambda:
1. With one single tuple: __call(x_y_z)__
2. With one multiple parameters: __call(x, y, z)__

You either need to make sure that you know how all callers (both inside and outside the codebase) will call the lambda --or-- you need to gracefully support _both_ the calling conventions.